### PR TITLE
Add UI buttons and modal dialog boxes.

### PR DESCRIPTION
**Pull request by 0mdc**
This changeset adds networked UI elements.
* Modal dialogue boxes.
* Buttons.

https://github.com/0mdc/siro_hitl_unity_client/assets/110583667/cde6936b-8e16-4b44-9230-93d37b267aa3

UI elements follow the [immediate mode GUI pattern](https://en.wikipedia.org/wiki/Immediate_mode_GUI). They are both keyframe consumers and message producers.

How buttons work:
1. Server creates a button named `X` and sends it to the client.
2. Client receives it, renders it, and stores whether it was clicked.
3. Client sends back `X`'s state.
4. Server checks whether `X` was clicked.

Dialogue boxes are simply containers of UI elements (right now only supporting buttons).

Depends on:
* https://github.com/facebookresearch/habitat-lab/pull/1956

### CODE REVIEW

File: Assets/Scripts/LoadProgressTracker.cs [hunk 1]
```diff
@@ -15,6 +15,9 @@ public class LoadProgressTracker
     uint _successCount = 0;
     uint _failureCount = 0;
 
+    // TODO: Refactor.
+    public bool _modalDialogueShown = false;
```
Comment (0mdc) [inline Assets/Scripts/LoadProgressTracker.cs:19]: This will be refactored in a later pass.

Comment (0mdc) [inline Assets/Scripts/LoadProgressTracker.cs:19]: These global states will be refactored in a later pass.

File: Assets/Scripts/UIElementDrawer.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            int windowWidth = width / 3;
+            int windowHeight = width / 5;
+            int windowXOffset = windowWidth - (windowWidth / 2);
+            int windowYOffset = windowHeight - (windowHeight / 2);
+
+            GUILayout.BeginVertical();
+            GUILayout.BeginHorizontal();
+            GUI.color = Color.white;
+            GUILayout.Window(
+                9, // TODO
```
Comment (0mdc) [inline Assets/Scripts/UIElementDrawer.cs:56]: ImGUI IDs will be taken care of in a later pass.

**Review by 0mdc — COMMENTED** [review]
Comments for reviewers.

**Review by aclegg3 — APPROVED** [review]

**Pull request merged**